pkgs/development/compilers/ghc/8.4.1.nix

```diff
@@ -0,0 +1,129 @@
+{ stdenv, targetPackages
+, buildPlatform, hostPlatform, targetPlatform
+, selfPkgs, cross ? null
+
+# build-tools
+, bootPkgs, alex, happy
+, autoconf, automake, coreutils, fetchgit, perl, python3
+
+, libiconv ? null, ncurses
+
+, # If enabled, GHC will be built with the GPL-free but slower integer-simple
+  # library instead of the faster but GPLed integer-gmp library.
+  enableIntegerSimple ? false, gmp ? null
+
+, version ? "8.4.20180115"
```

Comment:
Nitpick, not sure how important this is, but shouldn't this be `8.4.0.20180115`? (with the `.0` between `8.4` and the date).

Reply:
I don't really have an opinion, having taken `ghcHEAD` as basis..

Reply:
The difference is that HEAD is going to be released as 8.6.1, so that new version number will be higher than 8.5.x no matter what "x" is. In case of the 8.4.1 pre-release, the final version is going to be 8.4.1, which will be a version that's lower than what we currently use for the alpha. Arguably, 8.4.0.x would be a better choice for the alpha.
 
Having said that, I don't really care about this issue, because I don't think it makes any difference in practice. :-) People who use the 8.4.1-alpha tend to know what they are doing and won't be confused by this issue, I think.